src/requests/consumption.js

```diff
@@ -0,0 +1,744 @@
+					type: resource.type,
+				}
+			})
+			.filter((published_id) => published_id !== null && published_id !== undefined)
+		let scope = {
+			roles: [],
+			entityType: [],
+		}
+		let metaInformation = {
+			recommendedFor: [],
+		}
+		if (templateData?.targeting_criteria) {
+			templateData?.targeting_criteria.forEach((targeting) => {
+				const targeting_entity = targeting?.entity_targeting?.value
+
+				if (!scope.entityType.includes(targeting_entity)) scope.entityType.push(targeting_entity)
+				if (targeting?.roles) {
+					targeting.roles.forEach((role) => {
+						if (!scope.roles.includes(role.value)) scope.roles.push(role.value)
+						if (!metaInformation.recommendedFor.includes(role.label))
+							metaInformation.recommendedFor.push(role.label)
+					})
+				} else {
+					scope.roles = []
+					metaInformation.recommendedFor = []
+				}
+				targeting[targeting_entity].forEach((target) => {
+					if (scope[targeting_entity] == undefined) scope[targeting_entity] = []
+					if (metaInformation[targeting_entity] == undefined) metaInformation[targeting_entity] = []
+					metaInformation[targeting_entity].push(target.name)
+					scope[targeting_entity].push(target._id)
+				})
+			})
+		}
+
+		let template = {
+			scope,
+			metaInformation,
+			resourceType: [common.ROLLOUT_TYPE_PROGRAM],
+			language,
+			keywords,
+			concepts: templateData?.concepts ? templateData?.concepts : [],
+			components: [],
+			resourceDetails,
+			isAPrivateProgram: false,
+			isDeleted: false,
+			requestForPIIConsent: templateData?.requestForPIIConsent ? true : false,
+			rootOrganisations: [
+				templateData?.rootOrganisations ? templateData?.rootOrganisations : templateData?.organization?.id,
+			],
+			createdFor: [templateData?.createdFor ? templateData?.createdFor : templateData?.organization?.id],
+			deleted: false,
+			status: common.STATUS_ACTIVE.toLowerCase(),
+			owner: templateData?.created_by,
+			createdBy: templateData?.created_by,
+			updatedBy: templateData?.created_by,
+			externalId: generateExternalId(templateData?.title),
+			name: templateData?.title.trim(),
+			description: templateData?.description ? templateData?.description : '',
+			updatedAt: new Date(),
+			createdAt: new Date(),
+			__v: 0,
+			endDate: new Date(templateData?.end_date),
+			startDate: new Date(templateData?.start_date),
+		}
+
+		return { success: true, template }
+	} catch (error) {
+		console.error('Error in formatTemplate:', error.message)
+		return { success: false, error: error.message }
+	}
+}
+
+/**
+ * Publish the Program
+ * @name publishProjectTemplates
+ * @param {Object} programData - Program template data
+ * @returns {Object} - Response of Program creation
+ */
+const publishProgram = function (programData) {
+	return new Promise(async (resolve, reject) => {
+		const result = { success: false, templateId: null, error: null }
+		try {
+			// Format the template
+			let formattedTemplate = formatProgramTemplate(programData)
```

Comment:
@adithyadinesh0412 if the program is created already, its a edit after publish, then also as per this it will create new program and solution right, instead of updating it